great_expectations/core/result_format.py

```diff
@@ -1,10 +1,18 @@
 from __future__ import annotations
 
 import enum
+from typing import Final, Literal, Union
 
 
 class ResultFormat(str, enum.Enum):
     BOOLEAN_ONLY = "BOOLEAN_ONLY"
     BASIC = "BASIC"
     COMPLETE = "COMPLETE"
     SUMMARY = "SUMMARY"
+
+
+ResultFormatUnion = Union[
+    ResultFormat, dict, Literal["BOOLEAN_ONLY", "BASIC", "SUMMARY", "COMPLETE"]
+]
+
+DEFAULT_RESULT_FORMAT: Final = ResultFormat.SUMMARY
```

Comment:
Is this truly the default across all uses? I remember we had some inconsistencies

Reply:
I believe it is now. The reason I made the constant was because we did have inconsistencies before, and that seemed back. I'm going to hold this PR open a bit more before merging, so feel free to push back if there's something I'm not thinking of.